### PR TITLE
update stubs for couchbase 3.1.2

### PR DESCRIPTION
**Pull request by avsej**
Note that braces were moved to new line after I run cs-fix on the file

### CODE REVIEW

**Comment by isfedorov** [comment]
> Note that braces were moved to new line after I run cs-fix on the file

@avsej Probably you should rebase on master and perform `composer cs-fix` after that as there is exactly vice-versa cs-fixer rule to place braces on the same line like `foo() {}`

**Comment by avsej** [comment]
this is what I did. maybe the issue in docker image or something like that.

**Comment by isfedorov** [comment]
@avsej I've just tried to run `docker-compose -f docker-compose.yml run php composer cs-fix -d /opt/project/phpstorm-stubs` on your fork and file has been formatted as expected 
Screenshot 2021-05-13 at 17 21 19

**Comment by avsej** [comment]
If it is not too hard for you, could you please update my PR with your result? I'm using CentOS 8, and when I run `podman-compose` -- it raises errors, so I've used just image without compose and it seems like it does not pick up correct settings.

File: couchbase_v3/couchbase.php [hunk 1]
```diff
@@ -1516,7 +1532,7 @@ public function setName(string $name): CollectionSpec {}
 
     public function setScopeName(string $name): CollectionSpec {}
 
-    public function setMaxExpiry(float $ms): CollectionSpec {}
+    public function setMaxExpiry(long $ms): CollectionSpec {}
```
Comment (isfedorov) [inline couchbase_v3/couchbase.php:1535]: This is incorrect change as there is no such typehint

File: couchbase_v3/couchbase.php [hunk 2]
```diff
@@ -2827,7 +2843,7 @@ public function timeout(int $arg): GetAllReplicasOptions {}
      *
      *   `function decoder(string $bytes, int $flags, int $datatype): mixed`
      */
-    public function decoder(callable $arg): GetAllReplicasOptions {}
+    public function decoder(callable $arg): GetAllRepliacasOptions {}
```
Comment (isfedorov) [inline couchbase_v3/couchbase.php:2846]: Seems there is a typo in class name

File: couchbase_v3/couchbase.php (hunk 1 above)
Comment (avsej) [reply couchbase_v3/couchbase.php]: fixed

File: couchbase_v3/couchbase.php (hunk 2 above)
Comment (avsej) [reply couchbase_v3/couchbase.php]: thanks, I've fixed this too

**Comment by avsej** [comment]
@isfedorov, also I wonder if it is possible to swap directories like this:
* `couchbase` -> `couchbase_v2`
* `couchbase_v3` -> `couchbase`
Because V2 is not maintained anymore, and reached EOL, so people are getting confused if they install current version, but get stubs for old one. The old version V2 still makes sense if users are in process of migrating, but for new users it is really confusing.

**Comment by isfedorov** [comment]
> @isfedorov, also I wonder if it is possible to swap directories like this:

@avsej I think it's possible. Please do it in a separate PR but please note that `generate-stub-map` should be updated as well as it uses hardcoded `couchbase_v3`.

**Comment by avsej** [comment]
Ok, I will do it once this patch will be merged

**Comment by avsej** [comment]
@isfedorov I guess this PR is ready, or there is some pending changes or open questions?

**Comment by isfedorov** [comment]
@avsej PR seems to be ok to merge and will be merged soon. The only extra item for us is some of our internal tests that also use couchbase and they also should be updated.

**Pull request merged**